src/main/java/flaxbeard/steamcraft/block/BlockPipe.java

```diff
@@ -334,9 +334,12 @@ public void registerBlockIcons(IIconRegister ir) {
     @SubscribeEvent
     public void onBlockHighlight(DrawBlockHighlightEvent event) {
         Item equipped = event.player.getCurrentEquippedItem() != null ? event.player.getCurrentEquippedItem().getItem() : null;
-        if ((event.target.typeOfHit == MovingObjectPosition.MovingObjectType.BLOCK) && (event.player.worldObj.getBlock(event.target.blockX, event.target.blockY, event.target.blockZ) instanceof BlockPipe)
-                && (event.player.getCurrentEquippedItem() != null) && ((event.player.getCurrentEquippedItem().getItem() instanceof IPipeWrench && ((IPipeWrench) equipped).canWrench(event.player, event.target.blockX, event.target.blockY, event.target.blockZ)))) {
-            RayTracer.retraceBlock(event.player.worldObj, event.player, event.target.blockX, event.target.blockY, event.target.blockZ);
+        int x = event.target.blockX;
+        int y = event.target.blockY;
+        int z = event.target.blockZ;
+        if ((event.target.typeOfHit == MovingObjectPosition.MovingObjectType.BLOCK) && (event.player.worldObj.getBlock(x, y, z) instanceof BlockPipe)
+                && (equipped != null) && ((event.player.getCurrentEquippedItem().getItem() instanceof IPipeWrench && ((IPipeWrench) equipped).canWrench(event.player, x, y, z)))) {
```

Comment:
I placed equipped.getItem()  before instanceof; there I got a complaint.

On Mon, May 23, 2016 at 10:12 PM, Eli Clemente Gordillo Foster <
notifications@github.com> wrote:

> In src/main/java/flaxbeard/steamcraft/block/BlockPipe.java
> <https://github.com/Esteemed-Innovation/Flaxbeards-Steam-Power/pull/443#discussion_r64269516>
> :
>
> > @@ -334,9 +334,12 @@ public void registerBlockIcons(IIconRegister ir) {
> >      @SubscribeEvent
> >      public void onBlockHighlight(DrawBlockHighlightEvent event) {
> >          Item equipped = event.player.getCurrentEquippedItem() != null ? event.player.getCurrentEquippedItem().getItem() : null;
> > -        if ((event.target.typeOfHit == MovingObjectPosition.MovingObjectType.BLOCK) && (event.player.worldObj.getBlock(event.target.blockX, event.target.blockY, event.target.blockZ) instanceof BlockPipe)
> > -                && (event.player.getCurrentEquippedItem() != null) && ((event.player.getCurrentEquippedItem().getItem() instanceof IPipeWrench && ((IPipeWrench) equipped).canWrench(event.player, event.target.blockX, event.target.blockY, event.target.blockZ)))) {
> > -            RayTracer.retraceBlock(event.player.worldObj, event.player, event.target.blockX, event.target.blockY, event.target.blockZ);
> > +        int x = event.target.blockX;
> > +        int y = event.target.blockY;
> > +        int z = event.target.blockZ;
> > +        if ((event.target.typeOfHit == MovingObjectPosition.MovingObjectType.BLOCK) && (event.player.worldObj.getBlock(x, y, z) instanceof BlockPipe)
> > +                && (equipped != null) && ((event.player.getCurrentEquippedItem().getItem() instanceof IPipeWrench && ((IPipeWrench) equipped).canWrench(event.player, x, y, z)))) {
>
> What needed casting?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Esteemed-Innovation/Flaxbeards-Steam-Power/pull/443/files/7cffdf096a3adbf892bb16a53861d36d2de8f64a#r64269516>
>